pkgs/development/python-modules/jaxlib/default.nix

```diff
@@ -68,16 +68,17 @@ let
   effectiveStdenv = if cudaSupport then cudaPackages.backendStdenv else inputs.stdenv;
 
   meta = with lib; {
-    description = "JAX is Autograd and XLA, brought together for high-performance machine learning research";
+    description = "Source-built JAX backend. JAX is Autograd and XLA, brought together for high-performance machine learning research";
     homepage = "https://github.com/google/jax";
     license = licenses.asl20;
     maintainers = with maintainers; [ ndl ];
-    platforms = platforms.unix;
+
+    # Make this platforms.unix once Darwin is supported.
+    # The top-level jaxlib now falls back to jaxlib-bin on unsupported platforms.
     # aarch64-darwin is broken because of https://github.com/bazelbuild/rules_cc/pull/136
     # however even with that fix applied, it doesn't work for everyone:
     # https://github.com/NixOS/nixpkgs/pull/184395#issuecomment-1207287129
-    # NOTE: We always build with NCCL; if it is unsupported, then our build is broken.
```

Comment:
> # NOTE: We always build with NCCL; if it is unsupported, then our build is broken.

This is a peculiar comment, if the derivation depended on nccl and nccl was unsupported for the platform, the evaluation would fail saying exactly that: nccl is not supported on this platform. It also seems like the cpu-only jaxlib didn't actually have nccl in the inputs. Am I reading this right? CC @samuela

Reply:
i'm not entirely sure... my read is that `cudaSupport -> using nccl`, and the logic that was added just happened to forget about the CPU-only case.

cc @ConnorBaker who wrote this in https://github.com/NixOS/nixpkgs/commit/9bebd9e72d6b552fcfd3d1e6716eca6563944f42